lib/utils/HfbFrameView.hpp

```diff
@@ -0,0 +1,230 @@
+/*****************************************
+@file
+@brief Code for using the HfbFrameView formatted data.
+- hfbMetadata
+- HfbFrameView
+*****************************************/
+#ifndef HFBBUFFER_HPP
+#define HFBBUFFER_HPP
+
+#include "FrameView.hpp"   // for FrameView
+#include "Hash.hpp"        // for Hash
+#include "buffer.h"        // for Buffer
+#include "dataset.hpp"     // for dset_id_t
+#include "hfbMetadata.hpp" // for hfbMetadata
+#include "visUtil.hpp"     // for cfloat
+
+#include "gsl-lite.hpp" // for span
+
+#include <set>      // for set
+#include <stdint.h> // for uint32_t, uint64_t, uint8_t
+#include <string>   // for string
+#include <time.h>   // for timespec
+#include <tuple>    // for tuple
+#include <utility>  // for pair
+
+
+/**
+ * @brief The fields within the hfbBuffer.
+ *
+ * Use this enum to refer to the fields.
+ **/
+enum class hfbField { hfb };
+
+/**
+ * @class hfbFrameView
+ * @brief Provide a structured view of a hyperfine beam buffer.
+ *
+ * This class inherits from the FrameView base class and sets up a view on a hyperfine beam buffer
+ *with the ability to interact with the data and metadata. Structural parameters can only be set at
+ * creation, everything else is returned as a reference or pointer so can be
+ * modified at will.
+ *
+ * @todo This may want changing to use reference wrappers instead of bare
+ *       references.
+ *
```

Comment:
Is this still a valid todo? I don't see what the point of the change is, so it seems like it might be a legacy thing.